### ISSUE TITLE
warnings (BB #834)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#834]
This issue was migrated from bitbucket.
**Original Reporter:** PadreWoW
**Original Date:** 25.03.2015 07:14:08 GMT+0000
**Original Priority:** minor
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/834

<hr>

warning C4390:
here:
player.cpp

```
#!c++
if (!load)
DEBUG_LOG("Player::BindToInstance: %s(%d) is now bound to map %d, instance %d, difficulty %d", GetName(), GetGUIDLow(), save->GetMapId(), save->GetInstanceId(), save->GetDifficulty());
        return &bind;   <------ (Empty controlled statement)
```

warning C4065:
here:
spell.cpp

```
#!c++
switch (m_spellInfo->Id)
{
default:   <----------- (default with no case)
sLog.outError("Spell %u has SDC_BTM_NEEDS_SCRIPT_CHECK in bitMaskCondition (spell_dummy_condition), but hasn't been implemented, yet.", m_spellInfo->Id);
break;
 }
```

warning C4307:
here:
WaypointManager.cpp

```
#!c++
uint64 last_id = 0xFFFFFFFF + 1;   <-------- (integral constant overflow)
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 25.03.2015 07:15:12 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/834/#comment-16878087

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 27.03.2015 17:32:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/834/#comment-16975751

<hr>

C4390: its not a true warning, in this case, the compiler gives the hint cause DEBUG_LOG is a define.

how ever it should be fixed with this one:

```
#!c++

if (!load)
{
    DEBUG_LOG("Player::BindToInstance: %s(%d) is now bound to map %d, instance %d, difficulty %d", GetName(), GetGUIDLow(), save->GetMapId(), save->GetInstanceId(), save->GetDifficulty());
}
return &bind;
```

C4065: empty switch is maybe a missing implementation in this case, to fix warning to this

```
#!c++

/*switch (m_spellInfo->Id)
{
default:*/
sLog.outError("Spell %u has SDC_BTM_NEEDS_SCRIPT_CHECK in bitMaskCondition (spell_dummy_condition), but hasn't been implemented, yet.", m_spellInfo->Id);
/*break;
 }*/
```

but code should remain, maybe it will be used for something later

C4307: uint64 last_id = 0xFFFFFFFF + 1 is done on purpose, but maybe a other solution would be better.. how ever

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 31.03.2015 18:53:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/834/#comment-17065674

<hr>

Gonna take a look at these.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 01.04.2015 21:50:51 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/834/#comment-17098068

<hr>

Fix #834

→ <<cset c3cf7f9a14a0>>
